packages/fxa-settings/src/components/UnitRowSecondaryEmail/index.test.tsx

```diff
@@ -0,0 +1,237 @@
+    });
+  });
+
+  describe('multiple secondary emails set', () => {
+    it('renders as expected and with verified email text only present on the last verified', () => {
+      const emails = [
+        {
+          email: 'johndope@example.com',
+          isPrimary: true,
+          verified: true,
+        },
+        {
+          email: 'johndope2@example.com',
+          isPrimary: false,
+          verified: true,
+        },
+        {
+          email: 'johndope3@example.com',
+          isPrimary: false,
+          verified: true,
+        },
+        {
+          email: 'johndope4@example.com',
+          isPrimary: false,
+          verified: true,
+        },
+      ];
+      render(
+        <MockedCache account={{ emails }}>
+          <UnitRowSecondaryEmail />
+        </MockedCache>
+      );
+      const unitRowContents = screen.getAllByTestId('unit-row-content');
+      const unitRowHeaderValues = screen.getAllByTestId(
+        'unit-row-header-value'
+      );
+      const secondaryEmails = emails.filter((email) => !email.isPrimary);
+
+      expect(
+        screen.getAllByTestId('secondary-email-make-primary')
+      ).toHaveLength(3);
+      expect(
+        screen.getAllByTestId('secondary-email-default-content')
+      ).toHaveLength(1);
+
+      expect(unitRowContents[unitRowContents.length - 1]).toContainElement(
+        screen.getByTestId('secondary-email-default-content')
+      );
+
+      expect(unitRowHeaderValues).toHaveLength(3);
+      secondaryEmails.forEach((email, index) => {
+        expect(unitRowHeaderValues[index].textContent).toContain(email.email);
+      });
+    });
+
+    it('renders multiple unverified as expected', () => {
+      const emails = [
+        {
+          email: 'johndope@example.com',
+          isPrimary: true,
+          verified: true,
+        },
+        {
+          email: 'johndope2@example.com',
+          isPrimary: false,
+          verified: false,
+        },
+        {
+          email: 'johndope3@example.com',
+          isPrimary: false,
+          verified: true,
+        },
+        {
+          email: 'johndope4@example.com',
+          isPrimary: false,
+          verified: false,
+        },
+      ];
+      render(
+        <MockedCache account={{ emails }}>
+          <UnitRowSecondaryEmail />
+        </MockedCache>
+      );
+
+      expect(
+        screen.getAllByTestId('secondary-email-make-primary')
+      ).toHaveLength(1);
+      expect(
+        screen.getAllByTestId('secondary-email-default-content')
+      ).toHaveLength(1);
+      expect(
+        screen.getAllByTestId('resend-secondary-email-code-button')
+      ).toHaveLength(2);
+      expect(screen.getAllByTestId('unverified-text')).toHaveLength(2);
+    });
+  });
+
+  // describe('resendSecondaryEmailCode', () => {
+  //   it('displays a success message in the AlertBar', () => {
+
+  //   });
+
+  //   it('displays an error message in the AlertBar', () => {
+
+  //   });
+  // })
```

Comment:
You likely already know about this, but dropping a comment to be sure :)

Reply:
This is fxa-1648